src/show/views.py

```diff
@@ -76,3 +78,32 @@ def post(self, request, pk):
             show.save()
 
         return success_response(self.serializer_class(show, context={"request": request}).data)
+
+
+class AddToListView(generics.GenericAPIView):
+
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def post(self, request, pk):
+        data = json.loads(request.body)
+        list_ids = data.get("list_ids")
+        if not Show.objects.filter(pk=pk):
+            return failure_response(f"show with id {pk} does not exist")
+
+        show = Show.objects.get(pk=pk)
+        success_lists = []
+        for lst_id in list_ids:
+            lst = Lst.objects.filter(pk=lst_id)
+            if lst:
+                lst = Lst.objects.get(pk=lst_id)
+                user_profile = Profile.objects.get(user=request.user)
+                is_owner = user_profile == lst.owner
+                is_collaborator = user_profile in lst.collaborators.all()
+                if not is_owner and not is_collaborator:
+                    continue
+                lst.shows.add(show)
+                success_lists.append(lst)
+
+        return success_response(
+            LstWithSimpleShowsSerializer(success_lists, many=True, context={"request": request}).data
```

Comment:
once you have the lists that you want, you need to use `prefetch_related` on the `filter` so that we only make one trip to the database (ex. get all the shows from all the lists in one database trip) instead of multiple trips (ex. for each list, go to the database to get the shows --> meaning we make `n` trips to the database for `n` lists)